contracts/ArchToken.sol

```diff
@@ -0,0 +1,16 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.4;
+
+import "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+
+/// @title Archimedes Token
+/// @dev This is the contract for the Archimedes rewards token
+contract ArchToken is ERC20 {
+    // TODO set treasury address
+    address addressTreasury = msg.sender;
```

Comment:
I don't think we need that

We can transfer the treasury address with the constructor.
It is a one time thing, I don't see why we need a state or force deploying the contract from the Treasury address